packages/jest-matcher-utils/src/index.ts

```diff
@@ -412,7 +407,21 @@ const shouldPrintDiff = (actual: unknown, expected: unknown) => {
   return true;
 };
 
-function replaceMatchedToAsymmetricMatcher(
+export const replaceMatchedToAsymmetricMatcher = (
+  replacedExpected: unknown,
+  replacedReceived: unknown,
+  expectedCycles: Array<unknown>,
+  receivedCycles: Array<unknown>,
+): any => {
```

Comment:
It should also work without `any`, or?

```suggestion
): {replacedExpected: unknown; replacedReceived: unknown} => {
```